src/Data/Text/Utf8.hs

```diff
@@ -113,26 +137,8 @@ instance Hashable Text where
 instance NFData Text where
   rnf (Text (ByteArray !_) !_ !_) = ()
 
-newtype CodeUnitIndex = CodeUnitIndex
-    { codeUnitIndex :: Int
-    }
-    deriving stock (Eq, Ord, Show, Generic, Bounded)
-#if defined(HAS_AESON)
-    deriving newtype (Hashable, Num, NFData, AE.FromJSON, AE.ToJSON)
-#else
-    deriving newtype (Hashable, Num, NFData)
-#endif
-
--- TODO: Slow placeholder implementation until we can use text-2.0
-unpack :: Text -> String
-unpack = decodeUtf8 . unpackUtf8
-
--- TODO: Slow placeholder implementation until we can use text-2.0
-concat :: [Text] -> Text
-concat = pack . concatMap unpack
-
-null :: Text -> Bool
-null (Text _ _ len) = len == 0
+instance IsString Text where
+    fromString = pack
```

Comment:
Nit: for consistency, this should also use just two spaces for indentation.

Reply:
Addressed in 3b0e234